Panics on machine start and stop

I'm on a master-ish branch and I'm getting panics trying to do `docker-machine start foo`, `docker-machine stop foo`, etc. with `virtualbox` VM.

Can anyone bang on a Virtualbox with master (i.e. run a lot of `start` and `stop` in a shell loop) and see if they can reproduce please?

```console
$ ./docker-machine_darwin-amd64 start dev
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x1180a0]

goroutine 21 [running]:
runtime.panic(0x6a6d20, 0xaf52c4)
	/usr/src/go/src/pkg/runtime/panic.c:279 +0xf5
github.com/docker/machine/drivers/virtualbox.(*Driver).setupHostOnlyNetwork(0x208c462a0, 0x208be7e29, 0x3, 0x0, 0x0)
	/go/src/github.com/docker/machine/drivers/virtualbox/virtualbox.go:583 +0x8d0
github.com/docker/machine/drivers/virtualbox.(*Driver).Start(0x208c462a0, 0x0, 0x0)
	/go/src/github.com/docker/machine/drivers/virtualbox/virtualbox.go:377 +0xbd
github.com/docker/machine/libmachine.(*Host).Start(0x208c0e750, 0x0, 0x0)
	/go/src/github.com/docker/machine/libmachine/host.go:178 +0x59
github.com/docker/machine/libmachine.*Host.Start·fm(0x0, 0x0)
	/go/src/github.com/docker/machine/commands/commands.go:407 +0x38
github.com/docker/machine/commands.machineCommand(0x755fb0, 0x5, 0x208c0e750, 0x208bea2a0)
	/go/src/github.com/docker/machine/commands/commands.go:417 +0x493
created by github.com/docker/machine/commands.runActionForeachMachine
	/go/src/github.com/docker/machine/commands/commands.go:452 +0x2e4

goroutine 16 [chan receive]:
github.com/docker/machine/commands.runActionForeachMachine(0x755fb0, 0x5, 0x208c1a060, 0x1, 0x1)
	/go/src/github.com/docker/machine/commands/commands.go:453 +0x322
github.com/docker/machine/commands.runActionWithContext(0x755fb0, 0x5, 0x208c3a240, 0x0, 0x0)
	/go/src/github.com/docker/machine/commands/commands.go:481 +0x166
github.com/docker/machine/commands.cmdStart(0x208c3a240)
	/go/src/github.com/docker/machine/commands/start.go:10 +0x3e
github.com/codegangsta/cli.Command.Run(0x755fb0, 0x5, 0x0, 0x0, 0x76c410, 0xf, 0x819a90, 0x2a, 0x0, 0x0, ...)
	/go/src/github.com/docker/machine/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:113 +0xeb2
github.com/codegangsta/cli.(*App).Run(0x208c0e340, 0x208bf4000, 0x3, 0x3, 0x0, 0x0)
	/go/src/github.com/docker/machine/Godeps/_workspace/src/github.com/codegangsta/cli/app.go:156 +0xaea
main.main()
	/go/src/github.com/docker/machine/main.go:114 +0x639

goroutine 19 [finalizer wait]:
runtime.park(0x151f0, 0xaf9300, 0xaf7389)
	/usr/src/go/src/pkg/runtime/proc.c:1369 +0x89
runtime.parkunlock(0xaf9300, 0xaf7389)
	/usr/src/go/src/pkg/runtime/proc.c:1385 +0x3b
runfinq()
	/usr/src/go/src/pkg/runtime/mgc0.c:2644 +0xcf
runtime.goexit()
	/usr/src/go/src/pkg/runtime/proc.c:1445

goroutine 20 [syscall]:
os/signal.loop()
	/usr/src/go/src/pkg/os/signal/signal_unix.go:21 +0x1e
created by os/signal.init·1
	/usr/src/go/src/pkg/os/signal/signal_unix.go:27 +0x32
```

cc @sthulb @ehazlett @tianon @SvenDowideit @hairyhenderson

--- a/drivers/virtualbox/virtualbox.go
+++ b/drivers/virtualbox/virtualbox.go
@@ -582,6 +582,8 @@ func (d *Driver) generateDiskImage(size int) error {
 func (d *Driver) setupHostOnlyNetwork(machineName string) error {
 	hostOnlyCIDR := d.HostOnlyCIDR
 
+	// This is to assist in migrating from version 0.2 to 0.3 format
+	// it should be removed in a later release
 	if hostOnlyCIDR == "" {
 		hostOnlyCIDR = defaultHostOnlyCIDR
 	}

--- a/drivers/virtualbox/virtualbox.go
+++ b/drivers/virtualbox/virtualbox.go
@@ -27,7 +27,8 @@ import (
 )
 
 const (
-	isoFilename = "boot2docker.iso"
+	isoFilename         = "boot2docker.iso"
+	defaultHostOnlyCIDR = "192.168.99.1/24"
 )
 
 var (
@@ -96,7 +97,7 @@ func GetCreateFlags() []cli.Flag {
 		cli.StringFlag{
 			Name:   "virtualbox-hostonly-cidr",
 			Usage:  "Specify the Host Only CIDR",
-			Value:  "192.168.99.1/24",
+			Value:  defaultHostOnlyCIDR,
 			EnvVar: "VIRTUALBOX_HOSTONLY_CIDR",
 		},
 	}
@@ -579,7 +580,18 @@ func (d *Driver) generateDiskImage(size int) error {
 }
 
 func (d *Driver) setupHostOnlyNetwork(machineName string) error {
-	ip, network, err := net.ParseCIDR(d.HostOnlyCIDR)
+	hostOnlyCIDR := d.HostOnlyCIDR
+
+	if hostOnlyCIDR == "" {
+		hostOnlyCIDR = defaultHostOnlyCIDR
+	}
+
+	ip, network, err := net.ParseCIDR(hostOnlyCIDR)
+
+	if err != nil {
+		return err
+	}
+
 	nAddr := network.IP.To4()
 
 	dhcpAddr, err := getRandomIPinSubnet(network.IP)

